sr_robot_launch/src/sr_robot_launch/sr_ur_arm_unlock.py

```diff
@@ -0,0 +1,248 @@
+            program_running = self.call_arm_service(arm, "program_running", IsProgramRunning)
+            if (RobotMode.IDLE == robot_mode.robot_mode.mode or RobotMode.POWER_OFF == robot_mode.robot_mode.mode or
+                    SafetyMode.PROTECTIVE_STOP == safety_mode.safety_mode.mode or
+                    not program_running.program_running):
+                rospy.loginfo("Starting arm: %s", arm)
+                return True
+        return False
+
+    def clear_arms_popups(self):
+        for arm in self._arms:
+            self.call_arm_service(arm, "close_safety_popup", Trigger)
+        for arm in self._arms:
+            self.call_arm_service(arm, "close_popup", Trigger)
+
+    def load_arms_program_if_unloaded(self):
+        sleep_time = False
+        for arm in self._arms:
+            try:
+                headless_mode = rospy.get_param("/" + arm + "_sr_ur_robot_hw/headless_mode")
+            except KeyError:
+                headless_mode = False
+            if not headless_mode:
+                play_msg = self.call_arm_service(arm, "program_state", GetProgramState)
+                if "null" == play_msg.program_name:
+                    rospy.loginfo("Not in headless mode. Loading program: %s for arm: %s",
+                                  self._external_control_program_name, arm)
+                    resp = self.call_arm_service(arm, "load_program", Load,
+                                                 service_data=self._external_control_program_name)
+                    rospy.loginfo("%s", resp)
+                    sleep_time = True
+        return sleep_time
+
+    def start_arms_program_if_stopped(self):
+        sleep_time = False
+        for arm in self._arms:
+            try:
+                headless_mode = rospy.get_param("/" + arm + "_sr_ur_robot_hw/headless_mode")
+            except KeyError:
+                headless_mode = False
+            if not headless_mode:
+                play_msg = self.call_arm_service(arm, "program_state", GetProgramState)
+                if ProgramState.STOPPED == play_msg.state.state or ProgramState.PAUSED == play_msg.state.state:
+                    rospy.loginfo("Not in headless mode. Starting program: %s for arm: %s",
+                                  play_msg.program_name, arm)
+                    self.call_arm_service(arm, "play", Trigger)
+                    sleep_time = True
+            else:
+                rospy.loginfo("Headless mode active, resending robot program to arm: %s", arm)
+                self.call_arm_service(arm, "resend_robot_program", Trigger, dashboard=False)
+                self.call_arm_service(arm, "resend_robot_program", Trigger, dashboard=False)
+                sleep_time = True
+        return sleep_time
+
+    def release_arm(self):
+        try:
+            rospy.loginfo("Unlock arm(s) signal received. Checking arms.")
+            rospy.loginfo("Checking e-stops ...")
```

Comment:
"Checking arms and e-stops" maybe? just a visual thing as nothing is happening between those two lines

Reply:
the thinking was the first one signals that a signal has been received, and the second one says that we've started checks

Reply:
are you ok with it how it is? :)

Reply:
Fine for me